pkg/controller/control.go

```diff
@@ -1078,12 +1083,35 @@ func (c *Controller) metricsStart() {
 	go func() {
 		for {
 			time.Sleep(1 * time.Second)
+			c.metricsLock.Lock()
 			c.latestMetrics = c.metrics
 			c.metrics = &types.Metrics{}
+			c.metricsLock.Unlock()
 		}
 	}()
 }
 
-func (c *Controller) GetLatestMetics() *types.Metrics {
-	return c.latestMetrics
+func (c *Controller) GetLatestMetics() *ptypes.Metrics {
+	c.metricsLock.RLock()
+	defer c.metricsLock.RUnlock()
```

Comment:
Will this impact the rw performance? Almost No?

Reply:
I think no. I think it is a memory copy operation. But, I can benchmark and check it.

Reply:
Each r/w request received by the controller will call `recordMetrics`, it may affect the latency a little bit. You can have a simple benchmark if needed.

Reply:
Sure. Will update the result later.

Reply:
Why not just deep copy `c.metrics` here w/o read lock, then return the calculated copied?

Reply:
Need to calculate the average latency here, not just copy here.